src/main/java/org/apache/sling/pipes/internal/MovePipe.java

```diff
@@ -63,23 +67,38 @@ public boolean modifiesContent() {
             try {
                 Session session = resolver.adaptTo(Session.class);
                 if (session.itemExists(targetPath)){
-                    if (overwriteTarget && !isDryRun()) {
-                        logger.debug("overwriting {}", targetPath);
+                    if (overwriteTarget || orderBefore && !isDryRun()) {
```

Comment:
i'd clarify the test here with a set of parenthesis around `overwriteTarget  || orderBefore`